internal/utils/apply.go

```diff
@@ -76,8 +76,10 @@ func Apply(ctx context.Context, rcg genericclioptions.RESTClientGetter, opts *ru
 		changeSet.Append(cs.Entries)
 	}
 
-	if err := waitForSet(rcg, opts, changeSet); err != nil {
-		return "", err
+	if len(changeSet.Entries) > 0 {
```

Comment:
I'm not sure if this shouldn't be rather be made part of `ResourceManager#WaitForSet`.

Reply:
Hmmm I think if you want to wait you would want to wait for _something_ and not wait for whatever even if there is nothing there. This makes more sense here readability wise.

Ultimately I don't care where this check lives. 😊

Reply:
If I call `ResourceManager#WaitForSet` with an empty set and it doesn't immediately return (because there's nothing to wait for) this is a bug in that method IMHO.

Reply:
Why would you call Wait if there is nothing to wait for? :)

Reply:
That's like, I don't expect any visitors, but I open the door just in case. :D

Reply:
A function that behaves indeterministically if called with certain input is a recipe for disaster. Be liberal in what you accept. But I really don't mind in this specific case so just leaving this here as a general remark.

Reply:
I agree that we should look into the logic here and see where it hangs: https://github.com/fluxcd/pkg/blob/main/ssa/manager_wait.go But also the check in this PR is good, no need to instantiate a whole resource manager if there is nothing to wait for.

Reply:
I can do both if that's okay with you two?!